aiogram_dialog/widgets/kbd/calendar_kbd.py

```diff
@@ -41,13 +45,17 @@ class Calendar(Keyboard, ABC):
     def __init__(self,
                  id: str,
                  on_click: Union[OnDateSelected, WidgetEventProcessor, None] = None,
-                 when: Union[str, Callable] = None):
+                 when: Union[str, Callable] = None,
+                 locale: Text = Text('en_US')):
         super().__init__(id, when)
+        self._locale = locale
+        self.locale = None
         self.on_click = ensure_event_processor(on_click)
 
-    async def render_keyboard(self,
-                              data,
-                              manager: DialogManager) -> List[List[InlineKeyboardButton]]:
+    async def _render_keyboard(self,
+                               data,
+                               manager: DialogManager) -> List[List[InlineKeyboardButton]]:
+        self.locale = self._locale.render_text(data, manager)
```

Comment:
it could break in case of concurrent rendering. Do not use `self` for storing temporary and context specific data.